### DISCUSSION TITLE
Performance when using fetchMany with a nested where on two big tables

### DISCUSSION THREAD

**Discussion by jrmyio** [prisma/prisma#6722] [Q&A]
I am running into performance issues when trying to do a nested fetchMany. I have 2 tables which both have over a million records. A table `users (id, name, registered_at, group_id),` and a table `groups (id, name, active)`. I want to get all users that registered during a certain period and are part of a group that is active.

However, when doing a `fetchMany `on `users `with a nested where like: `{ registered_at: { gte : '2021-04-01 04:00:00' }, group: { active: 1 }}`  prisma will do an inner join on the whole users/groups table, which takes a ~minute, whereas the following query runs near instant: `SELECT * FROM (SELECT * FROM users WHERE registered_at >= '2021-04-01 04:00:00') as user_selection INNER JOIN groups ON users.group_id = user_groups.id WHERE groups.active = 1`.

Is there any workaround so that prisma can run such a query or are there any tweaks/workarounds one can do to get performance on multiple big tables while using nested queries? I cannot split this into two queries as the results of each query might return ten-thousands of rows.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
